[hashi_vault] allow for vault enterprise namespaces

drewmullen opened the pull request:
##### SUMMARY
vault included a namespace feature to segment secrets into 'mini vaults'. 

design:
pass namespace as parameter to Client class object, gets included as a header

users must upgrade to latest hvac, 0.7.0
##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
hashi_vault

##### ADDITIONAL INFORMATION
https://www.vaultproject.io/docs/enterprise/namespaces/index.html

bcoca committed:
unsafe cleanup

abadger committed:
Warn if empty string is present in synchronize's rsync_opts

drewmullen committed:
[hashi_vault] allow for vault enterprise namespaces

drewmullen committed:
[hashi_vault] allow for vault enterprise namespaces

drewmullen committed:
[hashi_vault] allow for vault enterprise namespaces

ianw committed:
Local test docs: correct tox shell command usage (#49350)

The ansible-test shell command doesn't have a --python option; you
have to specify the tox environment directly to the --tox command

Partially fixes: #49349

drewmullen commented:
https://github.com/ansible/ansible/blob/f74f7b03736a4b600e85ccad090f941595b47f6b/test/integration/targets/lookup_hashi_vault/playbooks/install_dependencies.yml#L7-L15

@mattclay are these HVAC version restrictions still necessary?

syncpak committed:
CPM plug configuration for WTI Power Devices (#47989)

* Add WTI OOB and PDU Device status, control and configuration module

* removed accidental file inclusions

* removed accidental file

* remove unneeded legacy files

* added the module cpm_plugconfig

syncpak committed:
CPM plugcontrol for WTI Power Devices (#47683)

* Add WTI OOB and PDU Device status, control and configuration module

* removed accidental file inclusions

* removed accidental file

* remove unneeded legacy files

* Added the new module cpm_plugcontrol to control the plugs on WTI Power devices

groggi committed:
Specify key to use while sorting permissions (#49126)

Without specifying the dictionary key to use while sorting it will fail
in Python 3 environments due to simplifying Python's rules for ordering
comparisons: https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons

XOmniverse committed:
Update intro_inventory.rst to show YAML ranges (#49265)

Adds a YAML example inventory listing a range of hosts to complement existing INI example.

acozine committed:
removes talk of hardcoded loading from plugin docs (#49383)

pillarsdotnet committed:
Do not document buggy behavior. (#49325)

In the pip module, umask should be specified as an octal *string*, not an octal (or decimal) *integer*.

Akasurde committed:
Add Action plugin example (#49290)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Filippo125 committed:
zabbix inventory: options to read per each host and set ansible_ssh_host (#44107)

pyr committed:
cloudstack: inventory: consider more keys optional (#49364)

bcoca committed:
Yaml inventory more tolerant (#48883)

* make yaml inv more tolerant to comments
* add tests for bad inventory processing

fixes #47254

rubentsirunyan committed:
zabbix: New zabbix_action module  (#49189)

felixfontein committed:
ACME: fix bug introduced in #49266, work around problem with Pebble (#49407)

carchi8py committed:
New options of na_ontap_aggregate (#48906)

* add new options for na_ontap_aggregate

* add gpl line

* remove dup option

* Put files in wrong directory

* change unit test to match the request from PR 48941

* Changed for review comments

* pep8

felixfontein committed:
ACME: add diff to acme_account, account_public_key to acme_account_facts, and general refactoring (#49410)

* Only one exit point.

* Refactoring account handling.

* Add diff support for acme_account.

* Insert public_account_key into acme_account_facts result and into acme_account diff.

* Add changelog.

anasbadaha committed:
Fix Issue:#47076 onyx_l2_interface parameter trunk_allowed_vlans tries to remove all existing vlans on a port (#49418)

Signed-off-by: Anas Badaha <anasb@mellanox.com>

cinek810 committed:
Dont' attempt to remove drive if ID is "None" (#49423)

bvitnik committed:
Update botmeta with maintainer info for XenServer related modules (#49424)

captainkerk committed:
Adds `redshift_cross_region_snapshots` module (#35527)

* add redshift_cross_region_snapshots module, unit tests.

* fix errors

* use ec2_argument_spec as the basis for the argument spec. fixed
metadata_version

* follow best practices by naming example tasks.

* code review changes

* fix linting errors

* Update version added

sdirbach committed:
Implement hide task arguments feature (#40176)

* Implement hide task arguments feature

* Add version_added field

* Include changelog file

fzwart committed:
ec2_instance: add placement_group support (#49286)

* add placement_group support

lpirl committed:
``make`` module: check return code (#47861)

akshay196 committed:
Support for pids_limit parameter in docker_container module (#49319)

* Support for pids_limit parameter in docker_container module

This add pids_limit parameter support in docker_container module

Fixes #43337

Signed-off-by: Akshay <akshay@localhost.localdomain>

* Add changelog for pids_limit parameter

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

* Remove unnecessary lines of code

The map is needed if the names are different.

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

* Update pids_limit option tests

It also run for docker-py < 1.10.0

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

Xaroth committed:
-Fix: The Ubuntu 14.04 image on Shippable runs an old version of python, causing issues with checking out bitbucket, adjust tests to compensate. (#49433)

-Add: Mention bitbucket dropping TLSv1 and TLSv1.1 support as per 01 Dec 2018, potentially causing issues with older clients.

rcarrillocruz committed:
Fix regex on ASA password prompt (#49438)

On Cisco ASAv on AWS, the prompt is 'Password', thus the privilege
elevation fails, we need to check upper and lower p.

gundalow commented on lib/ansible/plugins/lookup/hashi_vault.py:
```diff
@@ -8,7 +8,8 @@
 DOCUMENTATION = """
   lookup: hashi_vault
   author: Jonathan Davila <jdavila(at)ansible.com>
-  version_added: "2.0"
```
This string is meant to show the version of Ansible when the module was added, it shouldn't change

gundalow commented on lib/ansible/plugins/lookup/hashi_vault.py:
```diff
@@ -52,6 +53,9 @@
       default: True
+    namespace:
+      description: namespace where secrets reside
+      default: None
```
```suggestion
      default: None
      version_added: "2.8"
```

paneu committed:
nxos: Make sure config ends with a linefeed (#48465)

The config parser on NX-OS does not deal with configs that do not end
with a linefeed correctly which leads to various issues when the config
is loaded from startup-config upon reboot. Therefore, ensure that the
config returned is terminated by a linefeed.

gundalow committed:
Only one strategy_plugins page (#49442)

acozine committed:
Refines docs on creating action plugins (#49392)

* Refines docs on creating action plugins

Co-Authored-By: acozine <879121+acozine@users.noreply.github.com>

mattclay commented:
@drewmullen I'd ask @pilou-, who wrote those tests.

sdirbach committed:
Implement test case prefix to filter test cases (#40174)

* Implement test case prefix to filter test cases

* Cut line to not exceed 160 chars

* Replace tabs with spaces

* Add version_added field

* Include changelog file

mkrizek committed:
docs: try to clarify handlers a bit more (#49322)

* docs: try to clarify handlers a bit more
* docs: refines description of duplicate handler names

mikewiebe committed:
Fix issues related to track feature (#49454)

Akasurde committed:
VMware: Fix module usages in module_utils (#49421)

* VMware: Fix module usages in module_utils
* Skip test for Python 2.6 as SSL context is not available in Python 2.6

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Correct usages of custom error class (#48778)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

thyme-87 committed:
Fix typo in confirm new vault password message (#49468)

gundalow commented on lib/ansible/plugins/lookup/hashi_vault.py:
```diff
@@ -8,7 +8,8 @@
   author: Jonathan Davila <jdavila(at)ansible.com>
-  version_added: "2.0"
+  contributors: Drew Mullen <mullen.drew@gmail.com>
+  version_added: "2.1"
```
Why had this changed?

drewmullen replied:
the version? because we're adding new features. is that not right?

contributors? because i dont know the format well enough. ill remove it

gundalow replied:
Use this:
```yaml
version_added: "2.0"
author:
  - Jonathan Davila <jdavila(at)ansible.com>
  - Drew Mullen (@drewmullen) <mullen.drew@gmail.com>
```

`version_added` is the version of Ansible the module was added.

drewmullen replied:
thanks, mistake on my part

shatQ committed:
Extend functionality of filesystem module to support swap creation (#49456)

* Extended functionality of filesystem module to support swap creation

* Corrected documentation block

* Add a note to for version added

* was

pilou- commented on lib/ansible/plugins/lookup/hashi_vault.py:
```diff
@@ -52,6 +53,9 @@
       description: controls verification and validation of SSL certificates, mostly you only want to turn off with self signed ones.
       type: boolean
       default: True
+    namespace:
```
Mention that `namespace` requires hvac 0.7.0.

pilou- commented on lib/ansible/plugins/lookup/hashi_vault.py:
```diff
@@ -52,6 +53,9 @@
       description: controls verification and validation of SSL certificates, mostly you only want to turn off with self signed ones.
       type: boolean
       default: True
+    namespace:
```
Should not an `AnsibleError` be raised either when hvac doesn't handle namespace or when HashiCorp Vault service (<0.11) doesn't ?

drewmullen replied:
yeah, i am planning to add another try / catch to account for the older version tests (where namespaces isnt available)

pilou- reviewed (commented):
An integration test needs to be added in [test/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/](https://github.com/ansible/ansible/blob/devel/test/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/).

styks1111 committed:
v2 dnsimple api uses type vs record_type (#39301)

* v2 dnsimple api uses type vs record_type

https://developer.dnsimple.com/v2/zones/records/#listZoneRecords

* replace prio with priority

matonb committed:
Add module parameter to control Puppet agent noop switch (#43678)

* Puppet agent noop switch

    Add module parameter to control noop switch

    Current module overrides puppet.conf noop setting by forcing '--no-noop' switch  which may not be expected or desired.

    Add parameter to allow user control.

    noop: Undefined - Use agent configuration
    noop: true - Add --noop switch to cmd
    noop: false - Add --no-noop switch to cmd

Ansible check mode always runs with --noop switch

* Update lib/ansible/modules/system/puppet.py

Co-Authored-By: matonb <matonb@ltresources.co.uk>
Co-Authored-By: matonb <matonb@ltresources.co.uk>
Co-Authored-By: matonb <matonb@ltresources.co.uk>

sn4kebite committed:
rabbitmq_binding: Add support for state=absent (#48599)

* rabbitmq_binding: Add support for state=absent

* Add integration tests for rabbitmq_binding

* Update testcases

* Add changelog fragment

tomaszkiewicz committed:
Fix TargetDBInstanceIdentifier assignment in wrong place (issue #46689) (#47710)

richardcs committed:
Use System.Xml.XmlDocument.Load rather than Get-Content to avoid (#49206)

encoding errors when writing out XML

removing executable bit

refactor tests to handle sanity checking

linnil1 committed:
[Junos] Fix bug of setting user password (#44660)

* Add encrypted_password for junos

* Add readme for Junos user encrypted_password

* encrypted_password added in 2.8

mikewiebe committed:
Fix nxos_vxlan_vtep test (#49479)

Obirvalger committed:
Add apt_repo module (#49187)

samdoran committed:
Add better error when k=v syntax is used with YAML in tasks (#41754)

* Add error message for k=v and YAML in a single task

Find the correct line, column, and position for k=v errors since they are different than the position reported initially.

Document bug in quoting syntax check.

* Change tense or error message

Since the error still exists, switch to present tense rather than past tense.

* Remove double spaces after periods in error messages.

http://www.slate.com/articles/technology/technology/2011/01/space_invaders.html

* Add changelog fragment

* Add tests for new error message

* Fix tests

* Add clarifying comments to unit test

caphrim007 committed:
Fixes for examples in f5 modules (#49490)

So that they use provider instead of deprecated args method

samdoran committed:
Use IndexError in exception (#49488)

caphrim007 committed:
Various small fixes for f5 ansible modules (#49492)

abadger committed:
Revert "fix cache 'update' method to be 'mapping' compatible"

This reverts commit 68301f890addfe4ca8d3b57063e2f70fd21474c9.

caphrim007 committed:
Adds ASM policy fetch module (#49494)

drewmullen committed:
fix documentation entries and add prechecks for namespaces before instance creation

drewmullen committed:
fix documentation entries and add prechecks for namespaces before instance creation

s-hertel committed:
[s3_bucket] Handle error paginating object versions when bucket does not exist (#49396)

owenmann committed:
Update faq.rst: typo (#49491)

* Update faq.rst

<!--- Your description here -->
minor: typo

+label: docsite_pr

beeankha committed:
Change GB to GiB in some AWS modules (#49304)

drewmullen committed:
resolve merge conflict after rebase

caphrim007 committed:
Adds bigip_profile_analytics module (#49499)

DavidWittman committed:
Update BOTMETA ignore list for alternatives.py (#49489)

I'm getting way too many emails. Thank you!

robertdebock committed:
Fixes #49362 module dnf: install python3-dnf preferably over python2-dnf. (#49402)

* Create python3and2

Adding tests to see if dnf still works when python-2 and python-3 are installed.

* Update main.yml

Include the tests that run on python 2 and python 3, based on the package manager.

* Update dnf.py

Use python3-dnf by default, otherwise python2-dnf.

* Rename python3and2 to python3and2.yml

Rename.

it-praktyk committed:
Add win_psrepository (#48828)

* Add win_psrepository

* Small corrections

* Correct requiqurements - add NuGet

* Extend tests

* Post-review updates

* Post-review - updates 2

* Add empty RETURN section in documentation

* Add empty line at EOF

* Post-review updates 3

* Update tests to run in check_mode too

* Minor updates

* Correct installing/updating NuGet

* a few changes for review

* Remove copyrights

* Revert "Remove copyrights"

This reverts commit 2d687ec6a691d8d71358ae447e0fdc828976a767.

jamescassell committed:
become_method: make dzdo more like sudo (#47946)

dzdo is basically a drop-in replacement for sudo and supports the same
command line options.

There is no become_flags set for dzdo like there is for sudo, so users
will have to set that separately to have exactly the same functionality.

mattandes committed:
Fix win_nssm credentials quoting (#48761)

* Fix win_nssm credentials quoting

Fix credential quoting for win_nssm after changes to the way nssm command is invoked in Ansible 2.7.1.

* Updating nssm command to update credentials to use Argv-ToString to properly escape password variable.

* Adding changelog fragment for fix of #48728.

jeking3 committed:
azure_rm_image: fix creation of image with data disks (#49394)

dagwieers committed:
Fix typo

Fix a small typo

+label: docsite_pr

Akasurde committed:
AWS: Add support for available db types (#48703)

Add several database types which are supported by AWS in valid engine type.

Fixes: #48623

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

gundalow committed:
plugins/connection/saltstack.py is support: community (#49502)

drewmullen committed:
include requirements for namespaces in description

drewmullen committed:
include requirements for namespaces in description

drewmullen committed:
resolve merge conflict after rebase

drewmullen commented:
closing, will redo. want this feature included!
